unicodetools/src/main/java/org/unicode/text/UCD/TestUnicodeInvariants.java

```diff
@@ -326,6 +332,83 @@ protected String getFailure(int codepoint) {
         }
     }
 
+    private static void propertywiseLine(ParsePosition pp, String line, String file, int lineNumber)
+            throws ParseException {
+        pp.setIndex("Propertywise".length());
+        final UnicodeSet set = new UnicodeSet(line, pp, symbolTable);
+        if (set.hasStrings()) {
+            throw new ParseException(
+                    "Set should contain only single code points for property comparison",
+                    pp.getIndex());
+        }
+        expectToken("AreAlike", pp, line);
+        if (pp.getIndex() < line.length()) {
+            expectToken(",", pp, line);
+            expectToken("Except", pp, line);
+            expectToken(":", pp, line);
+        }
+        Set<String> excludedProperties = new HashSet<>();
+        excludedProperties.add("Name");
+        while (pp.getIndex() < line.length()) {
+            final int propertyNameStart = pp.getIndex();
+            scan(PATTERN_WHITE_SPACE, line, pp, false);
+            excludedProperties.add(line.substring(propertyNameStart, pp.getIndex()));
+            scan(PATTERN_WHITE_SPACE, line, pp, true);
+        }
+        final var iup = IndexUnicodeProperties.make(Settings.latestVersion);
+        final List<String> errorMessageLines = new ArrayList<>();
+        properties:
+        for (var p : UcdProperty.values()) {
+            final var property = iup.getProperty(p);
+            for (String alias : property.getNameAliases()) {
+                if (excludedProperties.contains(alias)) {
+                    continue properties;
+                }
+            }
+            final int first = set.charAt(0);
+            String p1 = property.getValue(first);
+            for (var range : set.ranges()) {
+                for (int c = range.codepoint; c <= range.codepointEnd; ++c) {
+                    String p2 = property.getValue(c);
```

Comment:
Aren't you comparing first with itself as well?
-->
```suggestion
                for (int c = range.codepoint; c <= range.codepointEnd; ++c) {
                    if (c == first) { continue; }
                    String p2 = property.getValue(c);
```